examples/guides/se-restful-webservice/README.adoc

```diff
@@ -539,98 +462,80 @@ that we can use by simply setting the greeting to blank from
-                       final ServerResponse response) {
-        response
-                .status(Http.Status.OK_200)
-                .send();
-    }
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
 ----
-This method simply returns 200 so the client knows the service is up.
 --
-
-. Add a new `alive` method
+. Add a liveness check method.
 +
 --
+The new method returns a `HealthCheckResponse`. This will make it
+very easy to add our custom health check to the built-in ones already in the code.
+
 [source,java]
-// _include::0-19:{main-class}[tag=alive]
-----
-    private static void alive(final ServerRequest request,
-                        final ServerResponse response) {
-        /*
-         * Return 200 if the greeting is set to something non-null and non-empty;
-         * return 500 (server error) otherwise.
-         */
-        String greetServiceError = greetService.checkHealth(); //<1>
-        if (greetServiceError == null) {
-            response
-                    .status(Http.Status.OK_200) //<2>
-                    .send();
-        } else {
-            JsonObject returnObject = JSON.createObjectBuilder() //<3>
-                    .add("error", greetServiceError)
-                    .build();
-            response
-                    .status(Http.Status.INTERNAL_SERVER_ERROR_500) //<4>
-                    .send(returnObject);
-        }
+// _include::0-10:{greet-service}[tag=checkAlive,indent=0]
+----
+HealthCheckResponse checkAlive() {
+    HealthCheckResponseBuilder builder = HealthCheckResponse.builder()
+            .name("greetingAlive"); //<1>
+    if (greeting == null || greeting.trim().length() == 0) { //<2>
+        builder.down() //<3>
+               .withData("greeting", "not set or is empty");
+    } else {
+        builder.up(); //<4>
     }
+    return builder.build(); //<5>
+}
 ----
-<1> Delegates to `GreetService` to evaluate the liveness of the service -- in our case, is the greeting non-empty.
-<2> Replies with a simple 200 for the health case.
-<3> For the unhealthy case prepares a description of the problem...
-<4> ...and replies with a 500.
+<1> Use a builder to assemble the response, giving the health check a human-readable
+name.
+<2> Enforce that the greeting be non-empty and non-null in order for the
+greeting service to be considered alive.
+<3> For a null or empty greeting the response indicates that the service
+is _down_, in this case adding an explanation.
+<4> For a valid greeting the response says the service is _up_.
+<5> Either way, have the builder build the response.
 --
 
-. Add the new endpoints to the routing
-+
-In the `createRouting` method in `Main.java` insert these lines immediately
-after the `.register` invocations:
-+
---
+=== Revise `Main.java`
+We need to slightly modify the `createRouting` method by adding our custom health check to the `HealthSupportBuilder`.
```

Comment:
We need to modify the `createRouting` method slightly to add our custom health check to `HealthSupportBuilder`.

Reply:
Fixed